src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -3225,24 +3225,28 @@ _mongoc_cluster_run_opmsg_send (
       message_length += mcd_rpc_header_set_response_to (rpc, 0);
       message_length += mcd_rpc_header_set_op_code (rpc, MONGOC_OP_CODE_MSG);
 
-      mcd_rpc_op_msg_set_sections_count (rpc, cmd->payload ? 2u : 1u);
+      // Reserve one section for the body (kind 0).
+      BSON_ASSERT (cmd->payloads_count <= MONGOC_CMD_PAYLOADS_COUNT_MAX);
```

Comment:
```suggestion
      BSON_ASSERT (cmd->payloads_count < MONGOC_CMD_PAYLOADS_COUNT_MAX + 1u);
```
Recommend using `<` and keeping the +1/-1 for clarity.

Reply:
Scratch this suggestion: I think this assertion should simply be `cmd->payloads_count < MONGOC_CMD_PAYLOADS_COUNT_MAX` and be treated as no longer relevant to the body section reservation.

Reply:
I think this should remain as-is with `<=`. `cmd->payloads_count` may be equal to `MONGOC_CMD_PAYLOADS_COUNT_MAX` (defined to 2) if there are two kind 1 payloads (aka document sequences). Updated comment to clarify.

Reply:
Confused myself into thinking of the max as an exclusive bound rather than inclusive. LGTM.